docs/non-normative/http-migration.md

```diff
@@ -29,6 +29,9 @@ updated to the stable HTTP semantic conventions, they:
 - May drop the environment variable in their next major version and emit only
   the stable HTTP and networking conventions.
```

Comment:
I wonder if this whole section above is adding to the confusion. It was a good part of the blog post, but I wonder if the document here should just focus on the summary of changes?